h/routes.py

```diff
@@ -159,8 +159,7 @@ def includeme(config):  # pylint: disable=too-many-statements
     config.add_route(
         "api.group_member",
         "/api/groups/{pubid}/members/{userid}",
-        factory="h.traversal.GroupRequiredRoot",
-        traverse="/{pubid}",
+        factory="h.traversal.group_membership_api_factory",
```

Comment:
The group membership APIs were sharing the same factory (`GroupRequiredRoot`) with other group-related views like: web pages for viewing and editing groups, admin pages for editing and deleting groups, group GET and PATCH APIs, and the API for getting a list of a group's members. But a group is not the correct context object for group membership API requests: a group membership is. So changing the group membership API route to use its own separate factory.

We don't need the `traverse` argument: there's no need to h to use traversal and the Pyramid docs recommend against it except for specific use-cases that benefit from traversal, it's simpler without it (see below).

Reply:
Note that this route is used by four separate API views:

1. `add_member()` (`POST /api/groups/{pubid}/members/{userid}`): this is an existing API that's _not_ the remove-member API that this PR is about. This API should be unaffected by this PR.
2. `remove_member()` (`DELETE /api/groups/{pubid}/members/{userid}`): another existing API, this is the API that this PR will extend to allow users to remove members other than themselves.
3. `edit_member()` (`PATCH /api/groups/{pubid}/members/{userid}` with new roles in JSON body): this view doesn't exist yet but will be added in a future PR.
4. `get_member()` (`GET /api/groups/{pubid}/members/{userid}`): doesn't exist yet and may not be added because it isn't needed for the current project.